sdk/identity/azure-identity/inc/azure/identity/client_certificate_credential.hpp

```diff
@@ -55,7 +85,7 @@ namespace Azure { namespace Identity {
      * @param options Options for token retrieval.
      */
     explicit ClientCertificateCredential(
-        std::string const& tenantId,
+        std::string tenantId,
```

Comment:
Why is this change needed?

Going from const to non-const is non-breaking, so we are allowed to do so.

Reply:
It would be a breaking change to replace "`std::string const& fnParam`" with "`std::string& fnParam`", but replacing it with "`std::string fnParam`" is non-breaking and is unnoticeable for the user. Replacing "`std::string fnParam`" with "`std::string const& fnParam`" is also non-breaking and users won't notice it. Everything that was compiling before, will keep compiling.

Why we need this: previously, we would keep no copy of `tenantId` - so there was no need to copy it as a function arg - reference was enough: we'd read it and return. Now we do copy `tenantId` and store it in `ClientCredentialCore`. It is possible for `ClientCertificateCredential` to still have `tenantId` arg as const ref, but then when we copy, we'd invoke copy constructor. But if we have argument as by-value, we can `std::move` it.